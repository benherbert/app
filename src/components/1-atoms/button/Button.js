import React from 'react'
import classNames from 'classnames'

const Button = props => {
  const { ariaLabel, children, className, handleClick, isDisabled = false, type = 'button', qaId } = props
  const buttonClass = classNames(className, 'button')

  const renderButton = () => {
    const buttonAttributes = {
      type,
      className: buttonClass,
      'aria-label': ariaLabel,
      disabled: isDisabled,
      'data-qa': qaId,
      onClick: handleClick
    }

    return (
      <div className='wolf'>
        <button {...buttonAttributes}>{children}</button>
      </div>
    )
  }

  return renderButton()
}

export { Button }
