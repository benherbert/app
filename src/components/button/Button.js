import React from 'react'
import classNames from 'classnames'

function Button ({ ariaLabel, children, className, handleClick, isDisabled = false, type = 'button', qaId }) {
  const buttonClass = classNames(className, 'button')

  const renderButton = () => {
    const props = {
      type,
      className: buttonClass,
      'aria-label': ariaLabel,
      disabled: isDisabled,
      'data-qa': qaId,
      onClick: handleClick
    }

    return (
      <div className='wolf'>
        <button {...props}>{children}</button>
      </div>
    )
  }

  return renderButton()
}

export { Button }
