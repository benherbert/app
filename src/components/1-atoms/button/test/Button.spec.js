import React from 'react'
import { Button } from '../Button'
import renderer from 'react-test-renderer'

test('object assignment', () => {
  const data = { one: 1 }
  data['two'] = 2
  expect(data).toEqual({ one: 1, two: 2 })
})

test('null', () => {
  const n = null
  expect(n).toBeNull()
  expect(n).toBeDefined()
  expect(n).not.toBeUndefined()
  expect(n).not.toBeTruthy()
  expect(n).toBeFalsy()
})

it('renders correctly', () => {
  const tree = renderer.create(<Button>Facebook</Button>).toJSON()
  expect(tree).toMatchSnapshot()
})
