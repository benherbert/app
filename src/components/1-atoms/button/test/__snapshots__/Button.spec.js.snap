// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="button"
  disabled={false}
  type="button"
>
  Facebook
</button>
`;
